src/main/kotlin/app/revanced/patches/twitch/chat/antidelete/patch/ShowDeletedMessagesPatch.kt

```diff
@@ -21,25 +27,77 @@ class ShowDeletedMessagesPatch : BytecodePatch(
     listOf(
         SetHasModAccessFingerprint,
         DeletedMessageClickableSpanCtorFingerprint,
+        ChatUtilCreateDeletedSpanFingerprint
     )
 ) {
+    private fun createSpoilerConditionInstructions(register: String = "v0") = """
+        invoke-static {}, Lapp/revanced/twitch/patches/ShowDeletedMessagesPatch;->shouldUseSpoiler()Z
+        move-result $register
+        if-eqz $register, :no_spoiler
+    """
+
     override fun execute(context: BytecodeContext): PatchResult {
-        // Force set hasModAccess member to true in constructor
+        // Spoiler mode: Force set hasModAccess member to true in constructor
         with(DeletedMessageClickableSpanCtorFingerprint.result!!.mutableMethod) {
             addInstructions(
                 implementation!!.instructions.lastIndex, /* place in front of return-void */
                 """
+                    ${createSpoilerConditionInstructions()}
                     const/4 v0, 1
                     iput-boolean v0, p0, $definingClass->hasModAccess:Z
-                """
+                """,
+                listOf(ExternalLabel("no_spoiler", instruction(implementation!!.instructions.lastIndex)))
             )
         }
 
-        // Disable setHasModAccess setter
-        with(SetHasModAccessFingerprint.result!!.mutableMethod.implementation!!) {
-            addInstruction(0, BuilderInstruction10x(Opcode.RETURN_VOID))
+        // Spoiler mode: Disable setHasModAccess setter
+        with(SetHasModAccessFingerprint.result!!) {
+            mutableMethod.addInstruction(0, "return-void")
         }
```

Comment:
The [with scope function](https://kotlinlang.org/docs/scope-functions.html#with) generally makes sense if you want to make a semantically contextual switch in code. Here you use with to switch to the semantic context of "We are now working with the result of the `SetHasModAccessFingerprint` fingerprint". This usually makes sense if your code block involves lots of logic which is scoped to the new context, but for this code it is suggested to simply do the following:

```suggestion
        SetHasModAccessFingerprint.result!!.mutableMethod.addInstruction(0, "return-void")
```

You do not have to apply this in this PR. Also, while asserting non-null is easy and is commonly used through the patches, it is recommended to handle them, not required in this PR but in the future.